File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    N = len(numbers)
    cnt = 0
    
    def dfs(idx, total):
        nonlocal cnt
        if idx == N:
            if total == target:
                cnt += 1
            return
        
        for i in range(2):
            if i == 0:
                dfs(idx+1, total + numbers[idx])
                
            if i == 1:
                dfs(idx+1, total - numbers[idx])
                
    dfs(0, 0)
    
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    
    function dfs(idx, total) {
        if (idx === numbers.length) {
            return total === target ? 1:0 ;
        }
        
        const plusCnt = dfs(idx+1, total+numbers[idx])
        const minusCnt = dfs(idx+1, total-numbers[idx])
        
        return plusCnt+minusCnt
    }
    
    return dfs(0,0);
}